custom_components/rct_power/lib/api.py

```diff
@@ -51,14 +49,13 @@ class InvalidApiResponse(BaseApiResponse):
     cause: str
 
 
-ApiResponse = ValidApiResponse | InvalidApiResponse
-RctPowerData = dict[int, ApiResponse]
+type ApiResponse = ValidApiResponse | InvalidApiResponse
+type RctPowerData = dict[int, ApiResponse]
 
 
-def get_valid_response_value_or(
-    response: ApiResponse | None,
-    defaultValue: DefaultResponseValue,
-) -> ApiResponseValue | DefaultResponseValue:
+def get_valid_response_value_or[_R](
+    response: ApiResponse | None, defaultValue: _R
+) -> ApiResponseValue | _R:
```

Comment:
How about we keep the name descriptive?

```suggestion
def get_valid_response_value_or[_DefaultResponseValue](
    response: ApiResponse | None, defaultValue: _DefaultResponseValue
) -> ApiResponseValue | _DefaultResponseValue:
```

Reply:
Not sure that's necessary here. In the past when the TypeVar was reused for different functions it indeed made sense but this definition is so compact that I think just `_R` might be even more readable.

_I know some people would even advocate for removing the leading underscores._

Anyway, I don't have a strong opinion on this one, so let know me if which you prefer.

Reply:
Lacking a definitive style guide I'd be fine with keeping it short - especially since it's only a very local generic argument.